frontend/src/pages/RoadmapGraphPage.tsx

```diff
@@ -46,6 +45,33 @@ export const RoadmapGraphPage = () => {
     shallowEqual,
   );
 
+  // TODO: scrolling selected into view from BlockGraph doesn't work with this
+  const a = useRef<HTMLDivElement>(null);
+  const b = useRef<HTMLDivElement>(null);
+  const refs = useRef([a, b]);
+  const scrolling = useRef(0);
+
+  useEffect(() => {
+    const handleScroll = (e: any) => {
```

Comment:
`(e: any)`

Is there any good ways to give this one a type instead of any? I was struggling with the same in my code, couldn't figure out any ways to fix it.

Reply:
I don't know any good way either. The type for `scroll` event handler on div seems to be `(e: Event) => any` or `(this: HTMLDivElement, e: Event) => any`

And `Event` isn't generic over the target element, also `React.SyntheticEvent<T>` doesn't compile.

I think the cleanest way here would be type assertion on the event target ( and I should be using the `currentTarget` )
```ts
    const handleScroll = (e: Event) => {
      // ...
      const target = e.currentTarget as HTMLDivElement | null;
      if (target) {
        const other = target === a.current ? b : a;
        window.requestAnimationFrame(() =>
          other.current?.scroll({ left: target.scrollLeft }),
        );
      }
    };
```
or picking the correctly typed object based on the event target
```ts
      const target = e.currentTarget;
      const self = (target === a.current ? a : b).current;
      const other = (target === a.current ? b : a).current;
      if (self && other) {
        window.requestAnimationFrame(() =>
          other.scroll({ left: self.scrollLeft }),
        );
      }
```


One way the *types would match* would be...
```ts
    function handleScroll(this: HTMLDivElement, e: Event) {
      // ...
      const other = this === a.current ? b : a;
      window.requestAnimationFrame(() =>
        other.current?.scroll({ left: this.scrollLeft }),
      );
    }
    const elements = [a.current, b.current];
    elements.forEach((el) => el?.addEventListener('scroll', handleScroll));
```